pkg/types/builder.go

```diff
@@ -31,34 +31,21 @@ import (
 // NewBuilder returns a new Builder.
 func NewBuilder(pkg *types.Package) *Builder {
 	return &Builder{
-		Package:  pkg,
-		genTypes: map[string]*types.Named{},
+		Package: pkg,
 	}
 }
 
 // Builder is used to generate Go type equivalence of given Terraform schema.
 type Builder struct {
 	Package *types.Package
 
-	genTypes map[string]*types.Named
+	genTypes []*types.Named
```

Comment:
Any reason to keep this as struct field?

Reply:
In every node of the recursion tree, the resulting types are added to that array. So, we need to be able to access the same array in each iteration of `buildResource`.